### DISCUSSION TITLE
Power automate / Teams "When someone responds to an adaptive card" trigger

### DISCUSSION THREAD

**Discussion by ebw44** [microsoft/AdaptiveCards#8595] [Q&A]
I'm trying to use the trigger "When someone responds to an adaptive card" in combination with the "Post adaptive card in a chat or channel", but when I click the "Submit" action button in teams (desktop, web or android) I get a "Unable to reach app. Please try again", and my response flow never gets trigger.
Both my post and response flow use the same 'Card Type ID' and they are both in my default environment (I have only one environment).
Anyone has any idea?

**Comment by dominikake** [comment]
Is this in Microsoft Teams?

**Reply by ebw44** [reply]
Yes, I forgot to give a bit of context. The adaptive card is created into a Teams group channel from power automate using the action "Post adaptive card in a chat or channel".
Then another flow contains the "When someone responds to an adaptive card" trigger witht the asme card type id.
The error I get inside team

**Reply by dominikake** [reply]
I haven't personally tried using that Trigger. What I read in the docs is it only works in the Productivity environment. What I have tried tho is the Action 'Post adaptive card and wait for a response'.

**Reply by ebw44** [reply]
I found a similar comments that you need to use a default environment. Which is my case (I have a single environment on this account).

**Reply by Sebeftw** [reply]
Hey @ebw44,
did you ever figure out what the issue was? I'm trying the same thing. Also in my default environment and I get nothing. :| 

Thanks!

**Reply by ebw44** [reply]
@Sebeftw, I never found a way to make this work.

**Reply by karlnr** [reply]
You may be receiving both the submit invoke event and the execute action's refresh invoke event as a pair on each submit action by the user.

I'm not familiar with power automate so this may be Teams bot specific, but I might suggest inspecting the order of invoke events received along with _all_ subsequent http responses returned with failing status codes. 

I can elaborate later as needed and if time allows of course, but enabling both in _tandem_ on the execute action based card _essentially_ reversed the order of events with respect to my expectations such that a refresh invoke event arrived a couple hundred milliseconds before the submit invoke action.

**Reply by karlnr** [reply]
Experience above is with a Teams-AI messaging bot btw.

Also worth noting I understood there to be a 60s timeout period implemented internally by the sdk  per docs as a buffer for certain actions and refresh trigger types, but I have yet to encounter this behavior though and so implement something similar internally.
